pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -1,259 +1,259 @@
+}
+
+void   _array_fill_cfloat(float complex c, t_ndarray arr)
+{
+    if (c == 0)
+        memset(arr.raw_data, 0, arr.buffer_size);
+    else
+        for (int32_t i = 0; i < arr.length; i++)
+            arr.nd_cfloat[i] = c;
+}
+
+
+void   _array_fill_cdouble(double complex c, t_ndarray arr)
+{
+    if (c == 0)
+        memset(arr.raw_data, 0, arr.buffer_size);
+    else
+        for (int32_t i = 0; i < arr.length; i++)
+            arr.nd_cdouble[i] = c;
+}
+
+/*
+** deallocation
+*/
+
+int32_t free_array(t_ndarray arr)
+{
+    if (arr.shape == NULL)
+        return (0);
+    free(arr.raw_data);
+    arr.raw_data = NULL;
+    free(arr.shape);
+    arr.shape = NULL;
+    free(arr.strides);
+    arr.strides = NULL;
+    return (1);
+}
+
+int32_t free_pointer(t_ndarray arr)
+{
+    if (arr.is_view == false || arr.shape == NULL)
+        return (0);
+    free(arr.shape);
+    arr.shape = NULL;
+    free(arr.strides);
+    arr.strides = NULL;
+    return (1);
+}
+
+/*
+** slices
+*/
+
+t_slice new_slice(int32_t start, int32_t end, int32_t step)
+{
+    t_slice slice;
+
+    slice.start = start;
+    slice.end = end;
+    slice.step = step;
+    return (slice);
+}
+
+t_ndarray array_slicing(t_ndarray arr, ...)
+{
+    t_ndarray view;
+    va_list  va;
+    t_slice slice;
+    int32_t start = 0;
+
+    view.nd = arr.nd;
+    view.type = arr.type;
+    view.type_size = arr.type_size;
+    view.shape = malloc(sizeof(int32_t) * arr.nd);
+    view.strides = malloc(sizeof(int32_t) * arr.nd);
+    memcpy(view.strides, arr.strides, sizeof(int32_t) * arr.nd);
+    view.is_view = true;
+    va_start(va, arr);
+    for (int32_t i = 0; i < arr.nd ; i++)
+    {
+        slice = va_arg(va, t_slice);
+        view.shape[i] = (slice.end - slice.start + (slice.step - 1)) / slice.step; // we need to round up the shape
+        start += slice.start * arr.strides[i];
+        view.strides[i] *= slice.step;
+    }
+    va_end(va);
+    view.raw_data = arr.raw_data + start * arr.type_size;
+    view.length = 1;
+    for (int32_t i = 0; i < view.nd; i++)
+            view.length *= view.shape[i];
+    return (view);
+}
+
+/*
+** assigns
+*/
+
+void        alias_assign(t_ndarray *dest, t_ndarray src)
+{
+    /*
+    ** copy src to dest
+    ** allocate new memory for shape and strides
+    ** setting is_view to true for the garbage collector to deallocate
+    */
+
+    *dest = src;
+    dest->shape = malloc(sizeof(int32_t) * src.nd);
+    memcpy(dest->shape, src.shape, sizeof(int32_t) * src.nd);
+    dest->strides = malloc(sizeof(int32_t) * src.nd);
+    memcpy(dest->strides, src.strides, sizeof(int32_t) * src.nd);
+    dest->is_view = true;
+}
+
+/*
+** indexing
+*/
+
+int32_t     get_index(t_ndarray arr, ...)
+{
+    va_list va;
+    int32_t index;
+
+    va_start(va, arr);
+    index = 0;
+    for (int32_t i = 0; i < arr.nd; i++)
+    {
+        index += va_arg(va, int32_t) * arr.strides[i];
+    }
+    va_end(va);
+    return (index);
+}
```

Comment:
Did this file change?

Reply:
didn't even touched it I don't know why it appears in the diff

Reply:
It might be a line ending thing due to windows vs fortran. You can get rid of the changes by doing:
`git checkout master -- pyccel/stdlib/ndarrays/ndarrays.c`